src/Symfony/Component/VarDumper/Dumper/CliDumper.php

```diff
@@ -477,4 +464,69 @@ protected function dumpLine($depth, $endOfValue = false)
         }
         parent::dumpLine($depth);
     }
+
+    /**
+     * Returns true if the stream supports colorization.
+     *
+     * Reference: Composer\XdebugHandler\Process::supportsColor
+     * https://github.com/composer/xdebug-handler
+     *
+     * @param mixed $stream A CLI output stream
+     *
+     * @return bool
+     */
+    private function hasColorSupport($stream)
+    {
+        if (!is_resource($stream) || 'stream' !== get_resource_type($stream)) {
+            return false;
+        }
+
+        if (DIRECTORY_SEPARATOR === '\\') {
+            return (function_exists('sapi_windows_vt100_support')
+                && @sapi_windows_vt100_support($stream))
+                || false !== getenv('ANSICON')
+                || 'ON' === getenv('ConEmuANSI')
+                || 'xterm' === getenv('TERM');
+        }
+
+        if (function_exists('stream_isatty')) {
+            return @stream_isatty($stream);
+        }
+
+        if (function_exists('posix_isatty')) {
+            return @posix_isatty($stream);
+        }
+
+        $stat = @fstat($stream);
+        // Check if formatted mode is S_IFCHR
+        return $stat ? 0020000 === ($stat['mode'] & 0170000) : false;
+    }
+
+    /**
+     * Returns true if the Windows terminal supports true color.
+     *
+     * Note that this does not check an output stream, but relies on environment
+     * variables from known implementations, or a PHP and Windows version that
+     * supports true color.
+     *
+     * @return bool
+     */
+    private function isWindowsTrueColor()
+    {
+        $result = strval(getenv('ANSICON_VER')) >= '183'
```

Comment:
thanks to PHP type juggling, this should be simplified to `183 <= getenv('ANSICON_VER')`

Reply:
Ah yes, that is a much better way of doing it.

Reply:
fixed in 923417122a